xpk.py

```diff
@@ -117,12 +117,160 @@
+                env:
+                - name: LD_LIBRARY_PATH
+                  value: /usr/local/nvidia/lib64
+              - name: maxtext-tcpx
+                image: {docker_image}
+                imagePullPolicy: Always
+                securityContext:
+                  privileged: true
+                ports:
+                    - containerPort: 6002
+                env:
+                  - name: REPLICATED_JOB_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['jobset.sigs.k8s.io/replicatedjob-name']
+                  - name: JOBSET_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['jobset.sigs.k8s.io/jobset-name']
+                  - name: JAX_COORDINATOR_ADDRESS
+                    value: "$(JOBSET_NAME)-$(REPLICATED_JOB_NAME)-0-0.$(JOBSET_NAME)"
+                  - name: NNODES
+                    value: "{args.num_slices}"
+                  - name: NODE_RANK
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['batch.kubernetes.io/job-completion-index']
+                  - name: USE_GPUDIRECT
+                    value: "tcpx"
+                  - name: GPUS_PER_NODE
+                    value: "{chips_per_vm}"
+                  - name: JAX_COORDINATOR_PORT
+                    value: "6002"
+                  - name: LD_LIBRARY_PATH
+                    value: /usr/local/nvidia/lib64
+                  - name: COMMAND
+                    value: "{command}"
+                  {args.env}
+                command:
+                  - "bash"
+                  - "-c"
+                  - |
+                    echo XPK Start: $(date) ; _sigterm() ( kill -SIGTERM $!;); trap _sigterm SIGTERM; (cd /deps && bash gpu_multi_process_run.sh) & PID=$!; while kill -0 $PID 2>/dev/null; do sleep 5; done; EXIT_CODE=$? ; echo XPK End: $(date); echo EXIT_CODE=$EXIT_CODE; echo Main app is done > /usr/share/maxtext/workload_terminated
+                volumeMounts:
+                  - name: nvidia-install-dir-host
+                    mountPath: /usr/local/nvidia/lib64
+                  - name: tcpx-nccl-plugin-volume
+                    mountPath: /usr/local/tcpx
+                  - name: tcpd-socket
+                    mountPath: /tmp
+                  - name: shared-memory
+                    mountPath: /dev/shm
+                  - name: workload-terminated-volume
+                    mountPath: /usr/share/maxtext
+                resources:
+                  limits:
+                    nvidia.com/gpu: {chips_per_vm}
+"""
+
 workload_delete_yaml = """apiVersion: jobset.x-k8s.io/v1alpha2
 kind: JobSet
 metadata:
   name: {args.workload}
-  annotations:
-    alpha.jobset.sigs.k8s.io/exclusive-topology: cloud.google.com/gke-nodepool # 1:1 job replica to node pool assignment
+  {annotation_config}
```

Comment:
Hmm, thinking about the deletion process more, currently we only support deleting TPU jobs, sad. And in order to support GPUs, we also need to add a new required argument for the device type.

---

I am thinking if there is a way to avoid this new required argument since it will break current user flows and shouldn't be needed in theory. 

@danielvegamyhre do you have any thoughts here. We want to delete a jobset just from its name. Do we need the annotation?

Reply:
I talked with @danielvegamyhre offline and he suggested that we delete the jobset by name using cli command or python sdk. Let's use CLI command for now since that aligns with our other command usage:

From Daniel:

You can delete a jobset by namespaced name
you'll just need to do it using a different kubectl command, or the python sdk
kubectl delete jobset <name> -n <namespace>

---

So the ask here is to move to `kubectl delete jobset {NAME} -n default`. Then minimizes complexity in workload delete code by a lot, yay.

The namespace we use is default. (https://github.com/google/xpk/blob/main/xpk.py#L1548). Feel free to create a const var for the namespace if you'd like.